crates/oxide/src/lib.rs

```diff
@@ -219,6 +219,36 @@ impl Scanner {
             self.files.extend(files);
             self.globs.extend(globs);
         }
+
+        // Find all `@source` globs that point to a directory. If so, promote the source to auto
+        // source detection instead.
+        if let Some(sources) = &mut self.sources {
+            for source in sources {
+                // If a glob ends with `**/*`, then we just want to register the base path as a new
+                // base.
+                if source.pattern.ends_with("**/*") {
+                    source.pattern = source.pattern.trim_end_matches("**/*").to_owned();
+                }
```

Comment:
1. `trim_end_matches` can be done unconditionally — no need to check `ends_with` first. I'd imagine the compiler is smart enough to omit the extra heap allocation in optimized builds.
2. `trim_end_matches` removes every occurrence which I think is fine. But the presence of the `ends_with` makes me think we wanted to do `strip_suffix` instead.